controllers/glance_controller.go

```diff
@@ -248,19 +248,20 @@ func (r *GlanceReconciler) reconcileDelete(ctx context.Context, instance *glance
 		}
 	}
 
-	/* TODO: if the CR is removed (Glance undeployed/cleaned up), we usually
-	* clean the resources created in the OpenStackControlPlane (e.g., we remove
-	* the database from mariadb, delete the service and the endpoints in
-	* keystone). We should delete the limits created for the Glance service,
-	* and do not leave leftovers in the controlplane.
-	 */
-
-	if instance.IsQuotaEnabled() {
+	// If the CR is removed (Glance undeployed/cleaned up), we usually
+	// clean the resources created in the OpenStackControlPlane (e.g.,
+	// we remove the database from mariadb, delete the service and the
+	// endpoints in keystone). We should delete the limits created for
+	// the Glance service, and do not leave leftovers in the ctlplane.
+
+	// do not attempt to remove limits if keystoneAPI are not available
+	_, err = keystonev1.GetKeystoneAPI(ctx, helper, instance.Namespace, map[string]string{})
```

Comment:
we already have finalizers related to both glance-{internal, external} endpoints and service, which means that we're not expecting keystone to be deleted before `Glance`.
Calling `GetKeystoneAPI` ensures the CR is still in the ctlplane.